### PR TITLE
Compose network error

### PR DESCRIPTION
**Pull request by gtardif**
**What I did**
* Detect when external network are not available when doing `docker compose up`
* Avid duplicate prefixing of network name with `myproject_`

**Related issue**
https://github.com/docker/compose-cli/issues/975
https://github.com/docker/compose-cli/issues/990

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: local/compose.go [hunk 1]
```diff
@@ -558,6 +558,10 @@ func (s *composeService) ensureNetwork(ctx context.Context, n types.NetworkConfi
 	_, err := s.apiClient.NetworkInspect(ctx, n.Name, moby.NetworkInspectOptions{})
 	if err != nil {
 		if errdefs.IsNotFound(err) {
+			if n.External.External {
+				return fmt.Errorf("Network %s declared as external, but could not be found", n.Name)
```
Comment (rumpl) [inline local/compose.go:562]: Shouldn't the linter fail on an error that starts with uppercase?

Comment (gtardif) [reply local/compose.go]: fixed. Indeed I thought linter did detect these things, it seems not (just double checked the linter is active in /local).

File: local/convergence.go [hunk 2]
```diff
@@ -240,8 +240,12 @@ func (s *composeService) runContainer(ctx context.Context, project *types.Projec
 		return err
 	}
 	id := created.ID
-	for net := range service.Networks {
-		name := fmt.Sprintf("%s_%s", project.Name, net)
+	for netName := range service.Networks {
+		network := project.Networks[netName]
+		name := fmt.Sprintf("%s_%s", project.Name, netName)
+		if network.External.External {
+			name = network.Name
```
Comment (ndeloof) [inline local/convergence.go:247]: IIRC, network name should be the one set in compose file (when set) even if this is a managed one (external=false). Would need to double check with docker-compose

Comment (gtardif) [reply local/convergence.go:247]: Indeed, tested with docker-compose, and if network name is specified in yaml, it's used as is, no prefix

Comment (gtardif) [reply local/convergence.go]: This network name setting needed to be fixed earlier, when creating/checking networks, in compose.go. Here we can just use the network.name that is set in the project.

File: local/compose.go [hunk 3]
```diff
@@ -59,7 +59,7 @@ func (s *composeService) Up(ctx context.Context, project *types.Project, detach
 		return err
 	}
 	for k, network := range project.Networks {
-		if !network.External.External && network.Name != "" {
+		if !network.External.External && network.Name == k {
```
Comment (gtardif) [inline local/compose.go:62]: ATM we can't tell if the network name was set by the user or set by default as the key value. Assuming if name = key that the name was not set by user. 
Will do a separate PR to update compose-go and let network name empty if not set by user, so we can differentiate the 2 cases

**Review by ndeloof — APPROVED** [review]
